src/UI/Component/MainControls/Factory.php

```diff
@@ -0,0 +1,271 @@
+	 *     clicking a different button. This does not apply to Buttons that direcly
+	 *     change the page.
+	 *
+	 *     Opening a slate by clicking an entry will close all other slates in the
+	 *     Mainbar. On desktop, slates open on the right hand of the Mainbar, between
+	 *     bar and content, thus "pushing" the content to the right, if there is not
+	 *     enough room.
+	 *
+	 *     If the content's width would fall below its defined minimum, the expanded
+	 *     slate is opened above (like in overlay, not "on top of") the content.
+	 *
+	 *     The slates height equals that of the Mainbar. Also, their position will
+	 *     remain static when the page is scrolled. A button to close a slate is
+	 *     rendered underneath the slate. It will  close all visible Slates and reset
+	 *     the states of all mainbar-entries.
+	 *
+	 *     When a tool (such as the help), whose contents are displayed in a slate,
+	 *     is being triggered, a special entry is rendered as first element of the
+	 *     Mainbar, making the available/invoked tool(s) accessible. Tools can be
+	 *     closed, i.e. removed from the Mainbar, via a Close Button. When the last
+	 *     Tool is closed, the tools-section is removed as well.
+	 *
+	 *   rivals:
+	 *     Tab Bar: >
+	 *       The Mainbar (and its components) shall not be used to substitute
+	 *       functionality available at objects, such as settings, members or
+	 *       learning progress. Those remain in the Tab Bar.
+	 *
+	 *     Content Actions: >
+	 *       "New item"-actions, the actions-menu (with comments, notes and tags),
+	 *       moving, linking or deleting objects and the like are NOT part of
+	 *       the Mainbar.
+	 *
+	 *     Personal Desktop: >
+	 *       The Personal Desktop provides access to services and tools and
+	 *       displays further information at first glance (e.g. the calendar).
+	 *       The Mainbar may reference those tools as well, but rather in form
+	 *       of a link than a widget.
+	 *
+	 *     Metabar: >
+	 *       Notifications from the system to the user, e.g. new Mail, are placed
+	 *       in Elements of the Metabar. The general direction of communication for
+	 *       the Mainbar is "user to system", while the direction is "system to user"
+	 *       with elements of the Metabar. However, navigation from both components
+	 *       can lead to the same page.
+	 *
+	 * context:
+	 *    - The Mainbar is used in the Standard Page.
+	 *
+	 * rules:
+	 *   usage:
+	 *     1: There SHOULD be a Mainbar on the page.
+	 *     2: If there is a Mainbar, it MUST be unique for the page.
+	 *
+	 *   composition:
+	 *     1: The bar MUST NOT contain items other than Bulky Buttons or Slates.
+	 *     2: The bar MUST contain at least one Entry.
+	 *     3: The bar SHOULD NOT contain more than five Entries.
+	 *     4: >
+	 *       Entries in the Mainbar MUST NOT be enhanced with counters or
+	 *       other notifications drawing the user's attention.
```

Comment:
"Tools must be Slates" is also a rule, right? Or does this appear somewhere else?

Reply:
Yes, tools will need to be implemented as slates to fit here. Tools is not an KS-entry in his own right, though, and currently we do not expect it to become one. Instead, a Slate is a tool when it is put in the according section of the Mainbar via `withToolEntry`.